src/resources/filters/quarto-pre/table-classes.lua

```diff
@@ -0,0 +1,88 @@
+-- table-classes.lua
+-- Copyright (C) 2020-2023 Posit Software, PBC
+
+-- helper for checking whether a single value is present in a table
+function contains(list, x)
+	for _, v in pairs(list) do
+		if v == x then return true end
+	end
+	return false
+end
+
+-- propagate table classes to the table's <table> element
+function tableTableTag() 
+  return {
+    Table = function(el)
+      if tcontains(el.attr.classes) then
+        local tblColwidths = el.attr.attributes[kTblColwidths]
+        el.attr.attributes[kTblColwidths] = nil
+        if tblColwidths ~= nil then
+          return pandoc.walk_block(el, {
+            Table = function(tbl)
+              tbl.attr.attributes[kTblColwidths] = tblColwidths
+              return tbl
+            end
+          })
+        end
+      end
+    end,
+  }
+end
+
+-- handle classes to pass to `<table>` element
+function tableClasses()
+
+  return {
+   
+    Table = function(tbl)
+      
+      -- if there is no caption then return tbl unchanged
+      if tbl.caption.long == nil or #tbl.caption.long < 1 then
+        return tbl
+      end
+
+      -- determine if we have any supplied classes, these should always begin with a `.` and
+      -- consist of alphanumeric characters
+      local caption =  tbl.caption.long[#tbl.caption.long]
+      local caption_str = pandoc.utils.stringify(caption)
```

Comment:
We should be using `parseTableCaption` here, like it's done in table-colwidth.lua